plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -188,6 +195,38 @@ public FileHiveMetastore(NodeVersion nodeVersion, HdfsEnvironment hdfsEnvironmen
         }
     }
 
+    @Override
+    public synchronized long acquireTableExclusiveLock(
+            HiveIdentity identity,
+            AcidTransactionOwner transactionOwner,
+            String queryId,
+            String dbName,
+            String tableName)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(dbName, tableName);
+
+        Long existingLockId = tableLockIds.get(schemaTableName);
+        ReentrantLock lock = existingLockId == null ? new ReentrantLock() : tableLocks.get(existingLockId);
+
+        // Wait for current lock holder to release it
+        lock.lock();
+
+        long lockId = new SecureRandom().nextLong();
+        tableLockIds.put(schemaTableName, lockId);
+        tableLocks.put(lockId, lock);
+
+        return lockId;
+    }
+
+    @Override
+    public synchronized void releaseTableLock(HiveIdentity identity, long lockId)
+    {
+        ReentrantLock lock = tableLocks.get(lockId);
```

Comment:
the lock should be removed from the map to avoid memory leak

(if we cannot do this, let's have a TODO comment, but i think we should be able to do this)